ui/views.py

```diff
@@ -154,6 +157,32 @@ def get_context_data(self, video, **kwargs):  # pylint: disable=arguments-differ
         return context
 
 
+class VideoDownload(VideoDetail):
+    """
+    Download a public video
+    """
+    def download(self, video):
+        """
+        Return the response as a file download
+        Returns:
+            HttpResponse: the response videofile content and a file attachment header
+        """
+        video_file = video.download
+        if not video_file:
+            raise Http404()
+        _, ext = splitext(video_file.s3_object_key)
+        video_bytes = requests.get(video_file.cloudfront_url, stream=True)
+        response = StreamingHttpResponse(
+            (chunk for chunk in video_bytes.iter_content(512 * 1024)),
+            content_type='video/mp4')
+        response['Content-Disposition'] = 'attachment; filename={}.{}'.format(slugify(video.title), ext)
+        return response
+
+    def get(self, request, *args, **kwargs):
+        video = get_object_or_404(Video, key=kwargs['video_key'], is_public=True)
```

Comment:
Do we need to do any additional permission check here? It looks like this will only work for cloudfront urls; are all cloudfront videos necessarily public?

Reply:
Nevermind, I missed the very obvious `is_public=True` here

Reply:
This is intended to download from cloudfront URL's; only public videos are allowed to be downloaded via these links.